thavalon-server/src/game/interactions.rs

```diff
@@ -44,47 +43,50 @@ impl ChannelInteractions {
 
     pub fn add_player(
         &mut self,
-        id: PlayerId,
+        name: String,
         incoming: mpsc::Receiver<Action>,
         outgoing: mpsc::Sender<Message>,
     ) {
-        self.inbox.insert(id, incoming);
-        self.outbox.insert(id, outgoing);
+        self.inbox.insert(name.clone(), incoming);
+        self.outbox.insert(name, outgoing);
```

Comment:
Do you need to clone here also?

Reply:
Nope! Since this is the last use of `name`, we can move it into the `outbox` instead of making a copy. The `add_player` function takes ownership of `name`, so it'll be deallocated at the end of it unless it's moved somewhere else like it is here